src/views/catalog/templatescatalog/hooks/useVmTemplatesWithAvailableSource.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { BOOT_SOURCE, useVmTemplates } from '@kubevirt-utils/resources/template';
+import {
+  getDataSource,
+  getPVC,
+  getTemplateBootSourceType,
+} from '@kubevirt-utils/resources/template/hooks/useVmTemplateSource/utils';
+
+export const useVmTemplatesWithAvailableSource = (
+  showOnlyAvailable = false,
+): useVmTemplatesWithAvailableSourceValues => {
+  const { templates, loaded } = useVmTemplates();
+
+  const [templatesWithSource, setTemplatesWithSource] = React.useState<V1Template[]>([]);
+  const [templatesWithSourceLoaded, setTemplatesWithSourceLoaded] = React.useState<boolean>(true);
+  const [error, setError] = React.useState<any>();
+
+  const initialAvailableTemplates = React.useMemo(
+    () =>
+      templates.filter((t) => {
+        const bootSource = getTemplateBootSourceType(t);
+        return bootSource?.type === BOOT_SOURCE.REGISTRY || bootSource?.type === BOOT_SOURCE.URL;
+      }),
+    [templates],
+  );
+
+  const promises = React.useMemo(
+    () =>
+      showOnlyAvailable &&
+      templatesWithSource.length === 0 &&
+      templates
+        .filter((template) => {
+          const bootSource = getTemplateBootSourceType(template);
+          return (
+            bootSource?.type === BOOT_SOURCE.PVC || bootSource?.type === BOOT_SOURCE.PVC_AUTO_UPLOAD
+          );
+        })
+        .map((template) => {
+          const bootSource = getTemplateBootSourceType(template);
+
+          if (bootSource.type === BOOT_SOURCE.PVC) {
+            const pvc = bootSource.source.pvc;
+            return getPVC(pvc.name, pvc.namespace).then(() => template);
+          }
+          const dataSource = bootSource.source.sourceRef;
+          return getDataSource(dataSource.name, dataSource.namespace).then((sourceRef: any) => {
```

Comment:
please use `V1beta1DataVolumeSourceRef` instead of `any` as type :)

Reply:
The returned type is not `V1beta1DataVolumeSourceRef` . its a `V1beta1DataVolume` which we don't have in the API
😢 
cc @yaacov

Reply:
https://github.com/kubevirt-ui/kubevirt-api/blob/0e189732504a3a49d570e83051d899493c472e84/containerized-data-importer/models/V1beta1DataVolume.ts